scrapy/spidermiddlewares/urllength.py

```diff
@@ -37,4 +38,10 @@ def _filter(request):
             else:
                 return True
 
-        return (r for r in result or () if _filter(r))
+        @_process_iterable_universal
+        async def process(result):
+            async for r in result or ():
+                if _filter(r):
+                    yield r
+
+        return process(result)
```

Comment:
I see you add this decorator to several middlewares, what if I have my own middlewares without process_iterable_universal? Will it work fine? Or will I have to add things to my middleware?

Reply:
UPDATE:

I tested it in one of my projects where I have some middleware added by me and it crashes. So it means users will have to update their middlewares.

Reply:
It's a tricky question and I'm going to add docs for this question and maybe do some more research when the current situation is documented. Old sync middlewares indeed cannot work with async callbacks at least with the code in this PR.